archive/j/java/DepthFirstSearch.java

```diff
@@ -0,0 +1,92 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class DepthFirstSearch {
+
+    public static void main(String[] args) {
+        if (args.length != 3 || args[0].isEmpty() || args[1].isEmpty() || args[2].isEmpty()) {
+            handleError();
+            return;
+        }
+        int matrixSize = args[0].split(",").length;
+        if (matrixSize != Math.pow(args[1].split(",").length, 2)){
+            handleError();
+            return;
+        }
+        List<Integer> vertices = Stream.of(args[1].split(","))
+                .map(String::trim)
+                .map(Integer::parseInt)
+                .toList();
+        int[][] matrix = new int[matrixSize][matrixSize];
+        int count = 0;
+        for (int i = 0; i < vertices.size(); i++) {
+            for (int j = 0; j < vertices.size(); j++) {
+                matrix[i][j] = Integer.parseInt(args[0].split(",")[count].trim());
+                count++;
+            }
+        }
+        DFSGraph dfsGraph = new DFSGraph(matrix, vertices);
+        int search = Integer.parseInt(args[2]);
+        System.out.println(dfsGraph.depthFirstSearch(0, 0, search));
+
+    }
+
+    private static void handleError() {
+        System.out.println("Usage: please provide a tree in an adjacency matrix form " +
+                "(\"0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, " +
+                "0\") together with a list of vertex values (\"1, 3, 5, 2, 4\") " +
+                "and the integer to find (\"4\")。");
```

Comment:
This character (`。`) needs to be removed from the string in order to get the tests to pass with Java 23